cli/command/image/push.go

```diff
@@ -48,10 +48,10 @@ func runPush(dockerCli *command.DockerCli, remote string) error {
 	requestPrivilege := command.RegistryAuthenticationPrivilegedFunc(dockerCli, repoInfo.Index, "push")
 
 	if command.IsTrusted() {
-		return trustedPush(ctx, dockerCli, repoInfo, ref, authConfig, requestPrivilege)
+		return trustedPush(ctx, dockerCli, repoInfo, ref.Familiar(), authConfig, requestPrivilege)
```

Comment:
I'd rather `trustedPush` did the "familiarization" itself. It's a bit scary to be passing `ref.Familiar()` to a function because it has the same type as `ref`, so it would be very easy to accidentally pass an unfamiliarized reference.